### ISSUE TITLE
Detect `asyncio-dangling-task` (`RUF006`) when discarding return value

### ISSUE THREAD

**Issue by tjkuson** [astral-sh/ruff#8863]
```python
import asyncio

for i in range(10):
    asyncio.create_task(foo)
```

triggers `asyncio-dangling-task` but

```python
import asyncio

for i in range(10):
    _ = asyncio.create_task(foo)
```

does not.

I would expect Ruff to trigger `asyncio-dangling-task` when the return value of `asyncio.create_task` is discarded, following the [rationale of the rule](https://docs.astral.sh/ruff/rules/asyncio-dangling-task/).

**Comment by zanieb** [comment]
This seems reasonable to me!

**Comment by allaboutevemirolive** [comment]
Hye, @tjkuson 

I'm fresh to Ruff's contributions. I try to duplicate your error by compile Ruff using `cargo build --release` and running the command on a python file.

```
./ruff check /home/nemesis/Documents/Github/test/python/game.py
```

game.py:

```py
import asyncio


async def some_coro(param):
    print(f"Executing coroutine with param: {param}")
    await asyncio.sleep(1)
    print(f"Coroutine with param {param} completed")


async def main():
    for i in range(10):
        asyncio.create_task(some_coro(param=i))


if __name__ == "__main__":
    asyncio.run(main())
```

I expected it would produce an error: `asyncio-dangling-task`, like you mentioned above, but it didn't.

The result is also the same for this snippet.

```py
_ = asyncio.create_task(some_coro(param=i))
``` 

Python Version:

```
$ python3 --version
Python 3.12.0
```

Local Machine:

```
Release 11 (bullseye) 64-bit
Kernel Linux 5.10.0-26-amd64 x86_64
Release 11 (bullseye) 64-bit
Memory: 7.5 GiB
Processor: Intel® Core™ i5-8250U CPU @ 1.60GHz × 8 
Graphics: Mesa Intel® UHD Graphics 620 (KBL GT2)
```

**Comment by charliermarsh** [comment]
@MichaReiser - Not intentional -- this rule doesn't look at _assignments_, it only looks at standalone expressions that include `asyncio.create_task(...)`, and then assumes that those are dangling by virtue of being a standalone expression statement.

**Comment by tjkuson** [comment]
@allaboutevemirolive I think you need to select the rule (it isn't enabled by default) using `--select RUF006`.

**Comment by asafamr-mm** [comment]
Hi,I gave it a try.
I'm new to rust (and the project codebase) and would love feedback. What do you think?

**Comment by charliermarsh** [comment]
Thanks @asafamr-mm!
